AirCasting/SearchAndFollow/CompleteScreen/Views/ChartSensor.swift

```diff
@@ -0,0 +1,69 @@
+// Created by Lunar on 23/05/2022.
+//
+
+import Foundation
+import SwiftUI
+
+protocol ChartSensor {
+    func clean(measurements: inout [SearchAndFollowChartViewModel.ChartMeasurement])
+}
+
+struct ChartSensorDefault: ChartSensor {
+    let name: String
+    var convertedName: EntriesSensor { return getEntriesSensor(using: name) }
+    typealias MeasurementType = SearchAndFollowChartViewModel.ChartMeasurement
+    
+    enum EntriesSensor: String {
+        case OpenAQ = "OpenAQ"
+        case AirBeam = "AirBeam"
+        case PurpleAir = "PurpleAir"
+        case undefined = ""
+    }
+    
+    func clean(measurements: inout [MeasurementType]) {
+        guard convertedName != .OpenAQ else { return }
+        guard let firstElement = measurements.reversed().first?.time else { return }
+        switch convertedName {
+        case .AirBeam:
+            // AB stands for AirBeam
+            clearABData(using: &measurements, hourToRemove: firstElement)
+        case .PurpleAir:
+            // PA stands for PurpleAir
+            clearPAData(using: &measurements, hourToRemove: firstElement)
+        case .undefined:
+            Log.info("Missing sensor name in the chart VM")
+        default:
+            Log.info("Missing sensor name in the chart VM")
+            return
+        }
+    }
+    
+    private func hoursForRemoval(using element: Date) -> Int { Calendar.current.component(.hour, from: element) }
+    
+    private func clearPAData(using updatedMeasurements: inout [MeasurementType], hourToRemove: Date) {
+        for (index, item) in updatedMeasurements.enumerated().reversed() {
+            if Calendar.current.component(.hour, from: item.time) == hoursForRemoval(using: hourToRemove) && Calendar.current.component(.minute, from: item.time) != 00 {
```

Comment:
Why we don't want to remove the measurement from the full hour?

Reply:
As I remember we have decided that using PurpleAir we want to have ex. those from 9:00 and exclude only those from above